src/main/scala/uk/gov/nationalarchives/tdr/api/service/TransferAgreementService.scala

```diff
@@ -1,48 +1,96 @@
+      ConsignmentMetadataRowWithName(
+        AllEnglishConfirmed, uuidSource.uuid, Some(consignmentId), Some(input.allEnglish.get.toString), time, userId),
+      ConsignmentMetadataRowWithName(
+        AppraisalSelectionSignOffConfirmed, uuidSource.uuid, Some(consignmentId), Some(input.appraisalSelectionSignedOff.get.toString), time, userId),
+      ConsignmentMetadataRowWithName(
+        CrownCopyrightConfirmed, uuidSource.uuid, Some(consignmentId), Some(input.allCrownCopyright.get.toString), time, userId),
+      ConsignmentMetadataRowWithName(
+        //initialOpenRecordsConfirmed field to be added to frontend
+        //temporarily default to true until value passed from frontend input
+        InitialOpenRecordsConfirmed, uuidSource.uuid, Some(consignmentId), Some("true"), time, userId),
+      ConsignmentMetadataRowWithName(
+        SensitivityReviewSignOffConfirmed, uuidSource.uuid, Some(consignmentId), Some(input.sensitivityReviewSignedOff.get.toString), time, userId)
+    )
   }
 
-  private def agreed(field: Option[Boolean]) = field.isDefined && field.get
+  private def convertDbRowsToTransferAgreement(consignmentId: UUID, rows: Seq[ConsignmentMetadataRowWithName]): TransferAgreement = {
+    val propertyNameToValue = rows.map(row => row.propertyName -> row.value.map(_.toBoolean)).toMap
+    TransferAgreement(consignmentId,
+      propertyNameToValue(PublicRecordsConfirmed),
+      propertyNameToValue(CrownCopyrightConfirmed),
+      propertyNameToValue(AllEnglishConfirmed),
+      propertyNameToValue(AppraisalSelectionSignOffConfirmed),
+      propertyNameToValue(InitialOpenRecordsConfirmed),
+      propertyNameToValue(SensitivityReviewSignOffConfirmed),
+      isAgreementComplete(propertyNameToValue)
+    )
+  }
 
-  private def isAgreementComplete(ta: TransferagreementRow): Boolean = {
-    val fields = List(ta.allcrowncopyright, ta.allenglish, ta.allpublicrecords, ta.appraisalselectionsignedoff, ta.sensitivityreviewsignedoff)
-    fields.forall(agreed)
+  def getTransferAgreement(consignmentId: UUID): Future[TransferAgreement] = {
+    consignmentMetadataRepository.getConsignmentMetadata(
+      consignmentId, transferAgreementProperties: _*).map(rows => convertDbRowsToTransferAgreement(consignmentId, rows))
+      .recover {
+      case nse: NoSuchElementException => throw InputDataException(s"Could not find metadata for consignment $consignmentId", Some(nse))
+      case e: SQLException => throw InputDataException(e.getLocalizedMessage, Some(e))
+    }
   }
 
-  private def dbRowToTransferAgreement(row: TransferagreementRow ): TransferAgreement = {
-    TransferAgreement(row.consignmentid,
-      row.allpublicrecords,
-      row.allcrowncopyright,
-      row.allenglish,
-      row.appraisalselectionsignedoff,
-      row.sensitivityreviewsignedoff,
-      Some(row.transferagreementid),
-      isAgreementComplete(row)
-    )
+  private def agreed(field: Option[Boolean]): Boolean = field.isDefined && field.get
```

Comment:
You can do `field.getOrElse(false)`, which is a bit neater because you don't have to use `.get` which looks unsafe even though it's OK here because of the `.isDefined`.

Hopefully this is a moot point once we've done [TDR-733](https://national-archives.atlassian.net/browse/TDR-733) and made all the fields in this table non-nullable.